numpy/core/tests/test_deprecations.py

```diff
@@ -148,15 +148,21 @@ class TestDatetime64Timezone(_DeprecationTestCase):
     naive fashion.
     """
     def test_string(self):
-        self.assert_deprecated(np.datetime64, args=('2000-01-01T00+01',))
-        self.assert_deprecated(np.datetime64, args=('2000-01-01T00Z',))
+        msg = "no explicit representation of timezones available for " \
+              "np.datetime64"
+        with pytest.warns(UserWarning, match=msg):
+            np.datetime64('2000-01-01T00+01')
+
+        with pytest.warns(UserWarning, match=msg):
+            np.datetime64('2000-01-01T00Z')
 
-    @pytest.mark.skipif(not _has_pytz,
-                        reason="The pytz module is not available.")
     def test_datetime(self):
-        tz = pytz.timezone('US/Eastern')
-        dt = datetime.datetime(2000, 1, 1, 0, 0, tzinfo=tz)
-        self.assert_deprecated(np.datetime64, args=(dt,))
+        msg = "no explicit representation of timezones available for " \
+              "np.datetime64"
+        with pytest.warns(UserWarning, match=msg):
+            t0 = np.datetime64('2023-06-09T12:18:40Z', 'ns')
+
+        t0 = np.datetime64('2023-06-09T12:18:40', 'ns')
```

Comment:
The tests no longer belong in this file, which is intended to test deprecations. Are the use cases covered in other tests?

Reply:
Ah good point! Moving the tests to `test_datetime.py`. 

From what I can tell, these tests are not covered in other places.